cmake/HPX_AddCompileTest.cmake

```diff
@@ -51,10 +51,13 @@ function(add_hpx_headers_compile_test category name)
 endfunction()
 
 function(add_hpx_lib_header_tests lib)
-  file(GLOB_RECURSE headers ${DO_CONFIGURE_DEPENDS} "${PROJECT_SOURCE_DIR}/include/hpx/*hpp")
+  file(GLOB_RECURSE headers ${DO_CONFIGURE_DEPENDS}
+      "${PROJECT_SOURCE_DIR}/include/hpx/*hpp")
+
   set(all_headers)
-  add_custom_target(tests.headers.${lib})
-  add_dependencies(tests.headers tests.headers.${lib})
+  add_custom_target(tests.headers.modules.${lib})
+  add_dependencies(tests.headers.modules tests.headers.modules.${lib})
+
```

Comment:
We should probably still have `tests.headers.modules` as a dependency of `tests.headers`.